mlflow/sklearn/utils.py

```diff
@@ -97,21 +127,178 @@ def _truncate_dict(d, max_key_length=None, max_value_length=None):
         should_truncate_key = (not key_is_none) and (len(str(k)) > max_key_length)
         should_truncate_val = (not val_is_none) and (len(str(v)) > max_value_length)
 
-        new_k = str(k)[:max_key_length] if should_truncate_key else k
+        new_k = _truncate_and_ellipsize(k, max_key_length) if should_truncate_key else k
         if should_truncate_key:
-            msg = "Truncated the key `{}`".format(k)
+            # Use the truncated key for warning logs to avoid noisy printing to stdout
+            msg = "Truncated the key `{}`".format(new_k)
             _logger.warning(msg)
 
-        new_v = str(v)[:max_value_length] if should_truncate_val else v
+        new_v = _truncate_and_ellipsize(v, max_value_length) if should_truncate_val else v
         if should_truncate_val:
-            msg = "Truncated the value `{}` (in the key `{}`)".format(v, k)
+            # Use the truncated key and value for warning logs to avoid noisy printing to stdout
+            msg = "Truncated the value of the key `{}`. Truncated value: `{}`".format(new_k, new_v)
             _logger.warning(msg)
 
         truncated[new_k] = new_v
 
     return truncated
 
 
+def _get_meta_estimators_for_autologging():
+    """
+    :return: A list of meta estimator class definitions
+             (e.g., `sklearn.model_selection.GridSearchCV`) that should be included
+             when patching training functions for autologging
+    """
+    from sklearn.model_selection import GridSearchCV, RandomizedSearchCV
+    from sklearn.pipeline import Pipeline
+
+    return [
+        GridSearchCV,
+        RandomizedSearchCV,
+        Pipeline,
+    ]
+
+
+def _is_parameter_search_estimator(estimator):
+    """
+    :return: `True` if the specified scikit-learn estimator is a parameter search estimator,
+             such as `GridSearchCV`. `False` otherwise.
+    """
+    from sklearn.model_selection import GridSearchCV, RandomizedSearchCV
+
+    parameter_search_estimators = [
+        GridSearchCV,
+        RandomizedSearchCV,
+    ]
+
+    return any(
+        [
+            isinstance(estimator, param_search_estimator)
+            for param_search_estimator in parameter_search_estimators
+        ]
+    )
+
+
+def _log_parameter_search_results_as_artifact(cv_results_df, run_id):
+    """
+    Records a collection of parameter search results as an MLflow artifact
+    for the specified run.
+
+    :param cv_results_df: A Pandas DataFrame containing the results of a parameter search
+                          training session, which may be obtained by parsing the `cv_results_`
+                          attribute of a trained parameter search estimator such as
+                          `GridSearchCV`.
+    :param run_id: The ID of the MLflow Run to which the artifact should be recorded.
+    """
+    with TempDir() as t:
+        results_path = t.path("cv_results.csv")
+        cv_results_df.to_csv(results_path)
```

Comment:
```suggestion
        cv_results_df.to_csv(results_path, index=False)
```

Should we remove the index column?

Reply:
Some users might want to save it to indicate the number of trials though